Request/ParamConverter/PsrServerRequestParamConverter.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory;
+use Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Converts HttpFoundation Request to PSR-7 ServerRequest using the bridge.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PsrServerRequestParamConverter implements ParamConverterInterface
+{
+    /**
+     * @var HttpMessageFactoryInterface
+     */
+    private $httpMessageFactory;
+
+    public function __construct(HttpMessageFactoryInterface $httpMessageFactory = null)
+    {
+        if (null === $httpMessageFactory) {
+            if (!class_exists('Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory')) {
+                throw new \RuntimeException('Install the PSR-7 bridge to inject a Psr\Http\Message\ServerRequestInterface instance in the controller.');
+            }
+
+            $httpMessageFactory = new DiactorosFactory();
+        }
+
+        $this->httpMessageFactory = $httpMessageFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(Request $request, ParamConverter $configuration)
+    {
+        $request->attributes->set($configuration->getName(), $this->httpMessageFactory->createRequest($request));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(ParamConverter $configuration)
+    {
+        if (null === $configuration->getClass()) {
+            return false;
+        }
+
+        return 'Psr\Http\Message\ServerRequestInterface' === $configuration->getClass();
```

Comment:
This is not supported by the bridge and it will not be easy to do: https://github.com/symfony/psr-http-message-bridge/pull/1/files#diff-7e5b3956b6ae2a5924719b9d4eecb8cdR43